src/components/shared/DownloadButton.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState, useEffect } from "react";
+import Translate from "../../utils/translate/translateService";
+import { sendAnalyticsEvent } from '../../utils/analyticsUtils';
+import { Button } from "semantic-ui-react";
+
+interface DownloadButtonProps {
+    buttonLabelKey?: string;
+    downloadLink?: string;
+    formLink?: string;
+    buttonSize?: "mini" | "tiny" | "small" | "large" | "huge";
+    buttonColour?: any;
+  }
+
+export default function DownloadButton(props: DownloadButtonProps) {
+    const [letDownload, setLetDownload] = useState(false);
+
+    const { buttonLabelKey = "DownloadCsv", 
+            downloadLink = "https://airtable.com/shraXWPJ9Yu7ybowM/tbljN2mhRVfSlZv2d?backgroundColor=blue&viewControls=on",
+            formLink = "https://docs.google.com/forms/d/e/1FAIpQLSdGd_wlq8YSyVPs2AOi1VfvxuLzxA8Ye5I3HkQwW_9yrumsCg/viewform",
+            buttonSize = "large",
+            buttonColour = "blue" } = props;
+    
+    useEffect(() => {
+        const doneSurvey = localStorage.getItem("DownloadFormCompleted");
+        if (doneSurvey) {
+            setLetDownload(true);
+        }
+    }, [])
+
+    const clickButton = async () => {
+        sendAnalyticsEvent({ category: 'Data Link Click', action: 'click', label: buttonLabelKey })
+        localStorage.setItem('DownloadFormCompleted', "true");
+        // Only allow users to access download form 3 mins after they clicked the original link
+        // to deter users from getting data without actually filling out the form
```

Comment:
Just curious, is this fool proof? Say a user were to forget to fill the form or the computer crashes (these are just rare edge cases) then they would get access to data without the form if I am right? Are we ok with that or will that just not happen?

Reply:
Yeah, we are okay with that. The foolproof solution would probably be to setup google authentication and wrangle with some google APIs to check for form completion. However frankly I don't think that's worth the effort.